Incorrect "case already handled" with tuples in features/recursive-patterns

**Version Used**: 68c9801, the latest commit on features/recursive-patterns

**Steps to Reproduce**:

Try to [compile the following code](https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLRIMaOQSwG4SoAOsMECAdsgDQwgEI5gCeAPgAIBMABAMICwAKADeQ7uO7sALNwCyACgD2wAFYQMMbooCUYiaMESj3ZAHccMDAAtu8pdW4AGbbsPHxB9+4xRkEWwBiiorcYMEOAPraIHpeXsBIUADWANyxcdw+frYAQlAI3MD5kdHpGRIJ0KllEgC+6fWCjUJc3EEhwtzNgq15BZ21QA==):

```c#
class C
{
    void M(object o)
    {
        switch ((o, 0))
        {
            case (Foo foo, _):
                break;
            case (Bar bar, _):
                break;
        }
    }
}

class Foo { }

class Bar { }
```

**Expected Behavior**:

The code compiles.

**Actual Behavior**:

The second `case` produces:

> error CS8120: The switch case has already been handled by a previous case.

I realize that the branch is just a prototype. So, my question is: is this currently expected to work? If this is a known problem due to the branch being a prototype, feel free to close this issue.

--- a/src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests2.cs
+++ b/src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests2.cs
@@ -1220,6 +1220,40 @@ class Program1
                 );
         }
 
+        [Fact, WorkItem(25591, "https://github.com/dotnet/roslyn/issues/25591")]
+        public void TupleSubsumptionError()
+        {
+            var source =
+@"class Program2
+{
+    public static void Main()
+    {
+        M(new Fox());
+        M(new Cat());
+        M(new Program2());
+    }
+    static void M(object o)
+    {
+        switch ((o, 0))
+        {
+            case (Fox fox, _):
+                System.Console.Write(""Fox "");
+                break;
+            case (Cat cat, _):
+                System.Console.Write(""Cat"");
+                break;
+        }
+    }
+}
+class Fox {}
+class Cat {}
+";
+            var compilation = CreatePatternCompilation(source);
+            compilation.VerifyDiagnostics(
+                );
+            var comp = CompileAndVerify(compilation, expectedOutput: @"Fox Cat");
+        }
+
         // PROTOTYPE(patterns2): Need to have tests that exercise:
         // PROTOTYPE(patterns2): Building the decision tree for the var-pattern
         // PROTOTYPE(patterns2): Definite assignment for the var-pattern

